File: JavaScript/bob.js
const hey = (message) => {
  message = message.trim();
  
  // Deal with Whitespaces 
  if(/(\\[ntr])|\s{3,}/g.test(message) || message === '') {
    return "Fine. Be that way!";
  }

  // Lowercase all the anocronyms in the string
  const anocronyms = ['OK', 'DMV'];
  let newMessage = message;
  
  anocronyms.forEach((a) => {
    newMessage = newMessage.replace(a, a.toLowerCase());
  }); 
  
  // Capital letters with question mark
  if(/[A-Z]+[?]$/g.test(newMessage)) {
    return "Calm down, I know what I'm doing!";
  }
  
  // Capital letters
  if(/[A-Z]{2,}/g.test(newMessage)) {
    return "Whoa, chill out!";
  }

  // Ends in a question mark
  if(/\?$/g.test(newMessage)) {
    return "Sure.";
  }

  return "Whatever."
};

const messages = [
  'Hey Bob, fancy going for tea?',
  'STOP DOING THAT!',
];

messages.forEach(message => {
  console.log(hey(message));
});

File: C#/Bob/Bob.cs
using System;
using System.Text.RegularExpressions;

public static class Bob
{
    public static string Response(string statement)
    {
        string s = statement.Trim();

        string pattern = @"(\\[ntr])|\s{3,}";
        var rg = new Regex(pattern);

        // Deal with Whitespaces 
        if (rg.IsMatch(s) || s == "") 
        {
            return "Fine. Be that way!";
        }

        // Lowercase all the anocronyms in the string
        string[] anocronyms = ["OK", "DMV", "NASA"];
        string newMessage = s;

        foreach(var a in anocronyms) 
        {
            newMessage = newMessage.Replace(a, a.ToLower());
        }

        // Capital letters with question mark
        pattern = @"[A-Z]+[?]$";
        rg = new Regex(pattern);

        if (rg.IsMatch(newMessage))
        { 
            return "Calm down, I know what I'm doing!";
        }

        // Capital letters
        pattern = @"[A-Z]{2,}";
        rg = new Regex(pattern);

        if (rg.IsMatch(newMessage)) 
        {
            return "Whoa, chill out!";
        }

        // Ends in a question mark
        pattern = @"\?$";
        rg = new Regex(pattern);
 
        if (rg.IsMatch(newMessage))
        {
            return "Sure.";
        }


        return "Whatever.";
    }
}